src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -118,9 +118,10 @@ private void LoadManifestsFromProvider(IWorkloadManifestProvider manifestProvide
                 using (Stream? localizationStream = readableManifest.OpenLocalizationStream())
                 {
                     var manifest = WorkloadManifestReader.ReadWorkloadManifest(readableManifest.ManifestId, manifestStream, localizationStream, readableManifest.ManifestPath);
-                    if (!_manifests.TryAdd(readableManifest.ManifestId, manifest))
+                    var manifestInfo = new WorkloadManifestInfo(manifest.Id, manifest.Version, readableManifest.ManifestDirectory, readableManifest.ManifestFeatureBand);
+                    if (!_manifests.TryAdd(readableManifest.ManifestId, (manifest, manifestInfo)))
```

Comment:
It should be very unusual for readableManifest.ManifestId to already be in _manifests, correct? That seems like an install failure.